pkg/objcache/objcache.go

```diff
@@ -119,19 +155,19 @@ func (c *Cache) Create(key string, size int64) (w io.WriteCloser, err error) {
 	// Function called on close which saves the object contents
 	// to the object cache.
 	onClose := func() error {
-		c.mutex.Lock()
-		defer c.mutex.Unlock()
 		if buf.Len() != int(size) {
```

Comment:
previous code of keeping `Lock` and `Unlock` together was better as separating it apart does not give much advantage as it's not doing any locking calls.